code/datums/components/shield.dm

```diff
@@ -183,6 +183,9 @@
 	var/obj/item/parent_item = parent
 	var/status_cover_modifier = 1
 
+	if(parent_item.integrity_failure && parent_item.obj_integrity <= parent_item.integrity_failure)
```

Comment:
Why do you check if integrity failure exists

Reply:
because it can be 0, in which case it aint supposed to work, though at that point it'd break ??